nomad/state/state_store.go

```diff
@@ -1006,6 +1016,16 @@ func (s *StateStore) DeleteJob(index uint64, namespace, jobID string) error {
 	txn := s.db.Txn(true)
 	defer txn.Abort()
 
+	err := s.DeleteJobTxn(index, namespace, jobID, txn)
+	if err == nil {
+		txn.Commit()
+	}
+	return err
+}
+
+// DeleteJobTxn is used to deregister a job, like DeleteJob,
+// but in a transcation.  Useful for when making multiple modifications atomically
```

Comment:
transaction

Reply:
Fixed in https://github.com/hashicorp/nomad/commit/5697cbb183342f3b5f8d1d32d70bafc08f1a91be